.github/workflows/main.yml

```diff
@@ -0,0 +1,57 @@
+name: todoapp
+run-name: ${{ github.actor }} - ${{ github.sha }}
+
+env:
+  PYTHON_VERSION: 3.8
+
+on:
+  push:
+    branches: ["main", "develop"]
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  python-ci:
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        working-directory: ./src
+
+    steps:
+
+    - uses: actions/checkout@v4
+      name: Checkout repository
+
+    - name: Set up Python ${{ env.PYTHON_VERSION }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ env.PYTHON_VERSION }}
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 coverage
+        pip install -r requirements.txt
+    - name: Lint
+      run: |
+        cd ../
+        flake8 . --show-source --statistics --exit-zero
+    - name: Check complexity
+      run: |
+        cd ../
```

Comment:
Similar to the previous comment, the `cd ../` command might be unnecessary. Ensure that the directory change is needed for the `flake8` complexity check. If not, it can be removed.